cmd/globaldns.go

```diff
@@ -0,0 +1,931 @@
+	}
+
+	provider, err := searchForGlobalDNSProvider(c, ctx.Args().First())
+	if nil != err {
+		return err
+	}
+
+	return outputMembers(ctx, c, provider.Members)
+}
+
+func globalDNSProviderCreate(ctx *cli.Context) error {
+	if ctx.NArg() == 0 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	name := ctx.Args().First()
+	providerType := ctx.String(argType)
+	rootDomain := ctx.String(argRootDomain)
+
+	if providerType == "" {
+		return errors.New("--type is required")
+	}
+
+	provider := &managementClient.GlobalDNSProvider{
+		Name:       name,
+		RootDomain: rootDomain,
+	}
+	switch providerType {
+	case route53Provider:
+		accessKey := ctx.String(argAwsAccessKey)
+		if accessKey == "" {
+			return errors.New("AWS access key is required for route53 type")
+		}
+		secretKey := ctx.String(argAwsSecretKey)
+		if secretKey == "" {
+			return errors.New("AWS secret key is required for route53 type")
+		}
+		provider.Route53ProviderConfig = &managementClient.Route53ProviderConfig{
+			AccessKey: accessKey,
+			SecretKey: secretKey,
+		}
+	case cloudflareProvider:
+		apiEmail := ctx.String(argCloudflareAPIEmail)
+		if apiEmail == "" {
+			return errors.New("API email is required for cloudflare type")
+		}
+		apiKey := ctx.String(argCloudflareAPIKey)
+		if apiKey == "" {
+			return errors.New("API key is required for cloudflare type")
+		}
+		provider.CloudflareProviderConfig = &managementClient.CloudflareProviderConfig{
+			APIEmail: apiEmail,
+			APIKey:   apiKey,
+		}
+	case alidnsProvider:
+		accessKey := ctx.String(argAlibabaCloudAccessKey)
+		if accessKey == "" {
+			return errors.New("Alibaba Cloud access key ID is required for alidns type")
+		}
+		secretKey := ctx.String(argAlibabaCloudSecretKey)
+		if secretKey == "" {
+			return errors.New("Alibaba Cloud access key secret is required for alidns type")
+		}
+		provider.AlidnsProviderConfig = &managementClient.AlidnsProviderConfig{
+			AccessKey: accessKey,
+			SecretKey: secretKey,
+		}
+	default:
+		return fmt.Errorf("unsupported provider type %q", providerType)
+	}
+
+	members, err := addMembersByNames(ctx, c, provider.Members, ctx.StringSlice(argMember), memberAccessTypeOwner)
+	if err != nil {
+		return err
+	}
+	provider.Members = members
+
+	if _, err = c.ManagementClient.GlobalDNSProvider.Create(provider); nil != err {
+		return err
+	}
+
+	fmt.Printf("Successfully created global DNS provider %q\n", name)
+	return nil
+}
+
+func globalDNSProviderUpdate(ctx *cli.Context) error {
+	if ctx.NArg() == 0 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	name := ctx.Args().First()
+
+	provider, err := searchForGlobalDNSProvider(c, name)
+	if err != nil {
+		return err
+	}
+
+	update := make(map[string]interface{})
+
+	if ctx.IsSet(argRootDomain) {
+		update["rootDomain"] = ctx.String(argRootDomain)
+	}
+
+	providerType := ""
+	if provider.Route53ProviderConfig != nil {
```

Comment:
Pull the switch up into these if blocks with an `else` for the error, `providerType` isn't used for anything else so no reason to check the same thing twice.